### PR TITLE
Design recommendations incorporated for 'Add/translate descriptions'

### PR DESCRIPTION
**Pull request by sharvaniharan**
Updated spinner item textSize.
Updated background color of card and activity

**Phab:** https://phabricator.wikimedia.org/T219651

### CODE REVIEW

File: app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml [hunk 1]
```diff
@@ -80,44 +80,79 @@
         <androidx.constraintlayout.widget.ConstraintLayout
             android:id="@+id/bottomButtonContainer"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:paddingBottom="8dp"
+            android:layout_height="50dp"
```
Comment (cooltey) [inline app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml]: The top of the `addDescriptionButton` has been cut off. The height of the container should be adjusted.

File: app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml [hunk 2]
```diff
@@ -80,44 +80,79 @@
...
+                android:layout_width="194dp"
+                android:layout_height="48dp"
                 android:layout_marginTop="8dp"
                 android:layout_marginBottom="8dp"
-                android:contentDescription="@string/view_next_random_article"
-                app:backgroundTint="?attr/paper_color"
-                app:borderWidth="0dp"
+                android:background="@drawable/button_shape_gray"
+                android:backgroundTint="?attr/colorAccent"
+                android:foreground="?attr/selectableItemBackgroundBorderless"
```
Comment (cooltey) [inline app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml]: Looks like it does not have the correct ripple animation and will show over the radius corners.

File: app/src/main/res/values-qq/strings.xml [hunk 3]
```diff
@@ -758,4 +758,6 @@
   <string name="text_input_dialog_ok_button_text">Positive action button text in pop-up dialog for text input.\n{{Identical|OK}}</string>
   <string name="text_input_dialog_cancel_button_text">Negative action button text in pop-up dialog for text input.\n{{Identical|Cancel}}</string>
   <string name="custom_date_picker_dialog_cancel_button_text">Negative action button text in date picker pop-up dialog.\n{{Identical|Cancel}}</string>
+  <string name="suggested_edits_add_translation">Label that prompts the user to translate a description of an article that does not yet have one.</string>
+  <string name="suggested_edits_edit_translation">Label that prompts the user to edit a translated description again.</string>
```
Comment (cooltey) [inline app/src/main/res/values-qq/strings.xml]: Can these have better names like `suggested_edits_add_translation_button` and `suggested_edits_edit_translation_button`?

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsActivity.kt [hunk 4]
```diff
@@ -13,16 +13,17 @@ import org.wikipedia.activity.SingleFragmentActivity
...
 
 class SuggestedEditsAddDescriptionsActivity : SingleFragmentActivity<SuggestedEditsAddDescriptionsFragment>() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        supportActionBar!!.elevation = if (intent.getSerializableExtra(EXTRA_SOURCE) == EDIT_FEED_TITLE_DESC) 8f else 0f
         supportActionBar!!.setDisplayHomeAsUpEnabled(true)
         supportActionBar!!.title = getString(if (intent.getSerializableExtra(EXTRA_SOURCE) == EDIT_FEED_TITLE_DESC)
             R.string.suggested_edits_add_descriptions else R.string.suggested_edits_translate_descriptions)
+        setStatusBarColor(ResourceUtil.getThemedAttributeId(this,R.attr.suggestions_background_color))
```
[the remark this answers is not in the document]
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsActivity.kt]: Done

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt [hunk 5]
```diff
@@ -128,6 +119,36 @@ class SuggestedEditsAddDescriptionsFragment : Fragment() {
...
+
+        backButton.setOnClickListener { previousPage() }
+        nextButton.setOnClickListener {
+            if (nextButton.drawable is Animatable) {
+                (nextButton.drawable as Animatable).start()
+            }
+            nextPage()
+        }
+        updateBackButton(0)
+        if (!TextUtils.isEmpty(topChild?.addedDescription)) addDescriptionImage!!.setImageDrawable(requireContext().getDrawable(R.drawable.ic_mode_edit_white_24dp))
```
[the remark this answers is not in the document]
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: Done

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt [hunk 6]
```diff
@@ -128,6 +119,36 @@ class SuggestedEditsAddDescriptionsFragment : Fragment() {
...
+        updateActionButton()
+    }
+
+    fun updateBackButton(pagerPosition: Int) {
+        backButton.isClickable = pagerPosition != 0
+        backButton.alpha = if (pagerPosition == 0) 0.31f else 1f
+    }
+
+    fun updateActionButton() {
+        if(!TextUtils.isEmpty(topChild?.addedDescription)) topChild?.showAddedDescriptionView(topChild?.addedDescription)
```
[the remark this answers is not in the document]
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: Done

File: app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml (hunk 2 above)
Comment (sharvaniharan) [reply app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml]: Right! done!

File: app/src/main/res/values-qq/strings.xml (hunk 3 above)
Comment (sharvaniharan) [reply app/src/main/res/values-qq/strings.xml]: Done

File: app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml (hunk 1 above)
Comment (sharvaniharan) [reply app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml]: Could you please give a screenshot @cooltey did this happen on a small screen?

Comment (cooltey) [reply app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml]: Screenshot_1557181594

It's on a large screen (1440 * 2880) emulator.

Comment (sharvaniharan) [reply app/src/main/res/layout/fragment_suggested_edits_add_descriptions.xml]: Thank you... done.

File: app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt [hunk 7]
```diff
@@ -5,6 +5,7 @@ import android.content.Context
 import android.net.Uri
 import android.text.TextUtils
 import android.view.View
+import android.widget.LinearLayout
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt]: Unused import

File: app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt [hunk 8]
```diff
@@ -84,6 +85,7 @@ class SuggestedEditsCardView(context: Context) : DefaultFeedCardView<SuggestedEd
         }
     }
 
+
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt]: Empty line

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsActivity.kt [hunk 9]
```diff
@@ -13,13 +13,13 @@ import org.wikipedia.activity.SingleFragmentActivity
 import org.wikipedia.analytics.SuggestedEditsFunnel
 import org.wikipedia.suggestededits.SuggestedEditsAddDescriptionsFragment.Companion.newInstance
 import org.wikipedia.util.FeedbackUtil
+import org.wikipedia.util.ResourceUtil
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsActivity.kt]: Unused import

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt [hunk 10]
```diff
@@ -159,21 +180,17 @@ class SuggestedEditsAddDescriptionsFragment : Fragment() {
         }
     }
 
-    private fun previousPage() {
+    fun previousPage() {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: can be private

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt [hunk 11]
```diff
@@ -159,21 +180,17 @@ class SuggestedEditsAddDescriptionsFragment : Fragment() {
...
         viewPagerListener.setNextPageSelectedAutomatic()
         if (addTitleDescriptionsItemPager.currentItem > 0) {
             addTitleDescriptionsItemPager.setCurrentItem(addTitleDescriptionsItemPager.currentItem - 1, true)
         }
+        updateActionButton()
     }
-
-    private fun nextPage() {
+    fun nextPage() {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: can be private

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt [hunk 12]
```diff
@@ -262,7 +279,7 @@ class SuggestedEditsAddDescriptionsFragment : Fragment() {
                 resetTitleDescriptionItemAdapter()
             }
             updateToLanguageSpinner(position)
-            updateBackButton(0)
+           updateBackButton(0)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: Please make the alignment.

File: app/src/main/res/layout-land/fragment_suggested_edits_add_descriptions_item.xml [hunk 13]
```diff
@@ -84,58 +84,46 @@
                     android:ellipsize="end"
                     tools:text="Lorem ipsum"/>
 
-                <androidx.constraintlayout.widget.ConstraintLayout
+                <LinearLayout
```
Comment (cooltey) [inline app/src/main/res/layout-land/fragment_suggested_edits_add_descriptions_item.xml]: If you are using the `LinearLayout`, you should not put the related `ConstraitLayout` attributes. (e.g. `layout_constraintStart_toStartOf`)

But I would recommend to keep using the `ConstraintLayout` anyway.

File: app/src/main/res/layout/fragment_suggested_edits_add_descriptions_item.xml [hunk 14]
```diff
@@ -52,6 +51,8 @@
                 android:id="@+id/headerView"
                 android:layout_width="match_parent"
                 android:layout_height="48dp"
+                android:paddingLeft="16dp"
```
Comment (cooltey) [inline app/src/main/res/layout/fragment_suggested_edits_add_descriptions_item.xml]: Would it be better if using `paddingStart` and `paddingEnd`?

File: app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt (hunk 7 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt]: Done

File: app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt (hunk 8 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt]: Done

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsActivity.kt (hunk 9 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsActivity.kt]: Done

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt (hunk 12 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: Done

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt (hunk 11 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: Done

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt (hunk 10 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsAddDescriptionsFragment.kt]: Done

File: app/src/main/res/layout-land/fragment_suggested_edits_add_descriptions_item.xml (hunk 13 above)
Comment (sharvaniharan) [reply app/src/main/res/layout-land/fragment_suggested_edits_add_descriptions_item.xml]: Done

File: app/src/main/res/layout/fragment_suggested_edits_add_descriptions_item.xml (hunk 14 above)
Comment (sharvaniharan) [reply app/src/main/res/layout/fragment_suggested_edits_add_descriptions_item.xml]: Done

**Pull request merged**